content/blog/2021/support-unknown.md

```diff
@@ -0,0 +1,348 @@
+at least browsers all agree 
+on how to handle this situation...
+at least on the surface.
+Behind the scenes 
+they are doing something slightly different,
+which we'll see in a minute.
+
+## Negating the unknown
+
+With both `selector()` and `container()`,
+we're testing for support of "wrapping" features:
+Selectors wrap around any number of declarations,
+and the `@container` query rule wraps around
+any number of selector rule-blocks.
+
+Since browsers ignore blocks of code 
+that they don't understand,
+these wrapping features often act as their own 
+`@supports` conditional test:
+
+```css
+/* only applied if the selector has support */
+.cat-item::marker {
+  content: '😻';
+}
+
+/* only applied if container queries have support */
+@container (min-width: 60ch) {
+  main { display: grid; }
+}
+```
+
+In these cases,
+it is much more useful to test
+for _lack of support_,
+and provide a fallback value.
+We want to use the negation syntax:
+
+```css
+@supports not selector(::marker) {
+  .cat-item { list-style: none; }
+  .cat-item::before { content: '😻';}
+}
+
+@supports not container(min-width: 60ch) {
+  @media (min-width: 50em) {
+    main { display: grid; }
+  }
+}
+```
+
+And this is where browsers disagree.
+We can see the result
+of their internal differences
+when interacting with unknown syntax.
+
+- Some browsers consider unknown syntax 
+  to mean _lack of support_, a value of `false`
+- Other browsers consider unknown syntax to be... unknown.
+  They return a value of `unknown`
+
+Those two values, 
+`false` and `unknown`,
+act the same when we're testing basic support --
+both of them act like `false`.
+But these values behave very differently 
+when we negate them:
+
+- `not false` is the same as `true`
+- `not unknown` is... still just `unknown` (acts like `false` again)
+
+You can't negate the unknown.
+
+## The path forward
+
+When it comes to testing support for new features,
+like Container Queries,
+we really want the ability to add new syntax,
+and then write negative tests of that syntax,
+and get the `not false` (`true`) result in old browsers.
+
+I know this is a lot of double-negatives,
+so let's look at it in context:
+
+```css
+@supports not container(min-width: 1em) {
+  /* we want old browsers to parse this code, */
+  /* so we need the negated test to be "true" */
+  /* "yes, we do not support container queries" */
+}
+```
+
+I opened an issue 
+with the CSS Working Group,
+where Oriol Brufau pointed out
+that the answer depends on
+[exactly how we ask the question][table].
+Specifically, we can get different results
+by adding/removing function arguments,
+or wrapping our query in parenthesis.
+
+[table]: https://github.com/w3c/csswg-drafts/issues/6175#issuecomment-826822088
+
+I've expanded on his table here
+to show the different results across browsers.
+You can see the live result from your current browser, 
+along with known results from the other major browsers.
```

Comment:
Am I supposed to be seeing something in the "Live Results" column?

![Screen Shot 2021-05-13 at 4 53 46 PM](https://user-images.githubusercontent.com/552316/118186585-e6844480-b40b-11eb-8289-a424b6de33e5.png)

Reply:
only if you get those styles working :)

Reply:
Ah, right -- that would require the custom stylesheet to be loaded 🤦